src/imagej/_java.py

```diff
@@ -1,8 +1,17 @@
+import logging
 from functools import lru_cache
 from typing import Callable
 
 from jpype import JArray, JClass, JObject
-from scyjava import jimport, jvm_started
+from scyjava import jimport, jstacktrace, jvm_started
+
+
+def log_exception(logger, exc):
+    if logger.isEnabledFor(logging.DEBUG):
+        jtrace = jstacktrace(exc)
+        if jtrace:
+            logger.debug(jtrace)
+
```

Comment:
It is a function of the private `_java` subsystem, so I think it's clear enough to write `_java.log_exception(...)`. Whereas `_java.log_java_exception` looks redundant to me.

Reply:
Hmm, good point. I agree.

What about adding a type hint/renaming `exc`, to prevent dummies like me from being confused?

Reply:
Also: `jstacktrace(exc)` returns None if the exception is not a Java one, so this function simply does nothing for Python exceptions. It doesn't crash.

Reply:
I wanted to do a type hint, but I don't think there is a Java exception base class in JPype? I'll double check, but I always have trouble finding it when I look.

Reply:
OK yeah, the base Java exception type, from a Python class perspective, is `_jpype._JException`. I.e.: it's a private class. Should we really be type-hinting that? If you think it's OK, I'll add it.

Reply:
We are trying to avoid explicit references to jpype though in downstream code, because it makes it harder to ever swap out the backend (e.g. jep).

Reply:
How about `log_jexception`? Too subtle?

Reply:
> We are trying to avoid explicit references to jpype though in downstream code, because it makes it harder to ever swap out the backend (e.g. jep).

How about typing on the actual java class? You can stringify the type hint to prevent it from prematurely starting up the JVM.
```
def log_exception(logger, exc: "jc.Exception"):
```
where `jc.Exception` returns `"java.lang.Exception"`

Reply:
> How about `log_jexception`? Too subtle?

`log_exception_only_if_its_from_java`

In all seriousness, I agree that you should just leave it. You're right that the module makes it clear.

Reply:
Oh wait, there _is_ a `jpype.JException`! Doh!

Reply:
With a704bfcfdeb89a6f5d93f76539756166e4867a9f, I added type hints for `log_exception`. I am still nervous about exposing jpype types in our downstream code, though. But `JException` is already used elsewhere... we can always change it later if/when scyjava has a functional second backend.

Reply:
@ctrueden did you see [this comment](https://github.com/imagej/pyimagej/pull/227#discussion_r992590162)?

Reply:
@gselzer Great idea! Is this what you had in mind? 75aeee2e5796aa2dd4a3a8d91902421145f02b1b

Reply:
Yup!